h2/stream.py

```diff
@@ -206,6 +216,8 @@ def stream_ended(self, previous_state):
         """
         Fires when a stream is cleanly ended.
         """
+        if previous_state is StreamState.HALF_CLOSED_LOCAL:
+            self.stream_closed_by = StreamClosedBy.RECV_END_STREAM
```

Comment:
Rather than check what the initial state is, it would be better to split this function up so that the right one is called based on the predicate state.

That is, rather than have `stream_ended` be called from two or more different `previous_state`, have two functions: one the regular `stream_ended` function as it was, and one that is only called from `HALF_CLOSED_LOCAL` and that does the `stream_closed_by` call *and then* calls the original `stream_ended` function.